## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 506835`
Commit: [`86dbe902519229e492221818b9015733f0e4d1b6`](https://github.com/NixOS/nixpkgs/commit/86dbe902519229e492221818b9015733f0e4d1b6) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/86dbe902519229e492221818b9015733f0e4d1b6..pull/506835/head))
Merge: [`98ab7b7c5b8ed0970ae54a2bbdec359dc7f0114f`](https://github.com/NixOS/nixpkgs/commit/98ab7b7c5b8ed0970ae54a2bbdec359dc7f0114f)

Logs: https://github.com/JamieMagee/nixpkgs-review-gha/actions/runs/23992612720


---
### `x86_64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>tests.home-assistant-component-tests.opendisplay</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 8 packages built:</summary>
  <ul>
    <li>python313Packages.epaper-dithering</li>
    <li>python313Packages.epaper-dithering.dist</li>
    <li>python313Packages.py-opendisplay</li>
    <li>python313Packages.py-opendisplay.dist</li>
    <li>python314Packages.epaper-dithering</li>
    <li>python314Packages.epaper-dithering.dist</li>
    <li>python314Packages.py-opendisplay</li>
    <li>python314Packages.py-opendisplay.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>tests.home-assistant-component-tests.opendisplay</summary>
<pre>2026-04-05 02:39:43.913 INFO     MainThread homeassistant.setup:setup.py:393 Setting up opendisplay
2026-04-05 02:39:43.915 INFO     SyncWorker_0 homeassistant.loader:loader.py:784 Loaded diagnostics from homeassistant.components.diagnostics
2026-04-05 02:39:43.917 INFO     MainThread homeassistant.setup:setup.py:393 Setting up diagnostics
2026-04-05 02:39:43.923 INFO     MainThread asyncio:base_events.py:1658 &lt;Server sockets=(&lt;asyncio.TransportSocket fd=28, family=2, type=1, proto=0, laddr=(&#x27;127.0.0.1&#x27;, 39529)&gt;,)&gt; is serving
2026-04-05 02:39:43.931 INFO     MainThread aiohttp.access:web_log.py:214 127.0.0.1 [05/Apr/2026:02:39:43 +0000] &quot;GET /api/diagnostics/config_entry/01KNDR5B93Q9JMYCZ73FH60KZ0 HTTP/1.1&quot; 200 3454 &quot;-&quot; &quot;Python/3.14 aiohttp/3.13.3&quot;
=============================== warnings summary ===============================
homeassistant/runner.py:176
  /build/source/homeassistant/runner.py:176: DeprecationWarning: &#x27;asyncio.DefaultEventLoopPolicy&#x27; is deprecated and slated for removal in Python 3.16
    class HassEventLoopPolicy(asyncio.DefaultEventLoopPolicy):  # type: ignore[name-defined,misc]

tests/conftest.py:154
  /build/source/tests/conftest.py:154: DeprecationWarning: &#x27;asyncio.set_event_loop_policy&#x27; is deprecated and slated for removal in Python 3.16
    asyncio.set_event_loop_policy(runner.HassEventLoopPolicy(False))

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
--------------------------- snapshot report summary ----------------------------
1 snapshot failed.
=========================== short test summary info ============================
FAILED tests/components/opendisplay/test_diagnostics.py::test_diagnostics - AssertionError: assert [+ received] == [- snapshot]
=================== 1 failed, 35 passed, 2 warnings in 3.25s ===================</pre>
</details>
</details>

---
### `aarch64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>tests.home-assistant-component-tests.opendisplay</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 8 packages built:</summary>
  <ul>
    <li>python313Packages.epaper-dithering</li>
    <li>python313Packages.epaper-dithering.dist</li>
    <li>python313Packages.py-opendisplay</li>
    <li>python313Packages.py-opendisplay.dist</li>
    <li>python314Packages.epaper-dithering</li>
    <li>python314Packages.epaper-dithering.dist</li>
    <li>python314Packages.py-opendisplay</li>
    <li>python314Packages.py-opendisplay.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-linux`</summary>
<details>
<summary>tests.home-assistant-component-tests.opendisplay</summary>
<pre>2026-04-05 02:39:35.771 INFO     MainThread homeassistant.setup:setup.py:393 Setting up opendisplay
2026-04-05 02:39:35.774 INFO     SyncWorker_0 homeassistant.loader:loader.py:784 Loaded diagnostics from homeassistant.components.diagnostics
2026-04-05 02:39:35.776 INFO     MainThread homeassistant.setup:setup.py:393 Setting up diagnostics
2026-04-05 02:39:35.784 INFO     MainThread asyncio:base_events.py:1658 &lt;Server sockets=(&lt;asyncio.TransportSocket fd=28, family=2, type=1, proto=0, laddr=(&#x27;127.0.0.1&#x27;, 43667)&gt;,)&gt; is serving
2026-04-05 02:39:35.795 INFO     MainThread aiohttp.access:web_log.py:214 127.0.0.1 [05/Apr/2026:02:39:35 +0000] &quot;GET /api/diagnostics/config_entry/01KNDR53ANK0MGY9TTCKC5K4SG HTTP/1.1&quot; 200 3455 &quot;-&quot; &quot;Python/3.14 aiohttp/3.13.3&quot;
=============================== warnings summary ===============================
homeassistant/runner.py:176
  /build/source/homeassistant/runner.py:176: DeprecationWarning: &#x27;asyncio.DefaultEventLoopPolicy&#x27; is deprecated and slated for removal in Python 3.16
    class HassEventLoopPolicy(asyncio.DefaultEventLoopPolicy):  # type: ignore[name-defined,misc]

tests/conftest.py:154
  /build/source/tests/conftest.py:154: DeprecationWarning: &#x27;asyncio.set_event_loop_policy&#x27; is deprecated and slated for removal in Python 3.16
    asyncio.set_event_loop_policy(runner.HassEventLoopPolicy(False))

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
--------------------------- snapshot report summary ----------------------------
1 snapshot failed.
=========================== short test summary info ============================
FAILED tests/components/opendisplay/test_diagnostics.py::test_diagnostics - AssertionError: assert [+ received] == [- snapshot]
=================== 1 failed, 35 passed, 2 warnings in 3.80s ===================</pre>
</details>
</details>

---
### `x86_64-darwin` (sandbox = relaxed)
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>python314Packages.py-opendisplay</li>
    <li>python314Packages.py-opendisplay.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 6 packages built:</summary>
  <ul>
    <li>python313Packages.epaper-dithering</li>
    <li>python313Packages.epaper-dithering.dist</li>
    <li>python313Packages.py-opendisplay</li>
    <li>python313Packages.py-opendisplay.dist</li>
    <li>python314Packages.epaper-dithering</li>
    <li>python314Packages.epaper-dithering.dist</li>
  </ul>
</details>

---
### `aarch64-darwin` (sandbox = relaxed)
<details>
  <summary>:white_check_mark: 8 packages built:</summary>
  <ul>
    <li>python313Packages.epaper-dithering</li>
    <li>python313Packages.epaper-dithering.dist</li>
    <li>python313Packages.py-opendisplay</li>
    <li>python313Packages.py-opendisplay.dist</li>
    <li>python314Packages.epaper-dithering</li>
    <li>python314Packages.epaper-dithering.dist</li>
    <li>python314Packages.py-opendisplay</li>
    <li>python314Packages.py-opendisplay.dist</li>
  </ul>
</details>

--- a/pkgs/development/python-modules/epaper-dithering/default.nix
+++ b/pkgs/development/python-modules/epaper-dithering/default.nix
@@ -1,0 +1,43 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  hatchling,
+  numpy,
+  pillow,
+  pytestCheckHook,
+}:
+
+buildPythonPackage (finalAttrs: {
+  pname = "epaper-dithering";
+  version = "0.6.3";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "OpenDisplay";
+    repo = "epaper-dithering";
+    tag = "python-v${finalAttrs.version}";
+    hash = "sha256-h84AgWJR8zVuwoz02A3U82yTOw4MSK9DjaxkYi0nWkM=";
+  };
+
+  sourceRoot = "${finalAttrs.src.name}/packages/python";
+
+  build-system = [ hatchling ];
+
+  dependencies = [
+    numpy
+    pillow
+  ];
+
+  nativeCheckInputs = [ pytestCheckHook ];
+
+  pythonImportsCheck = [ "epaper_dithering" ];
+
+  meta = {
+    description = "Dithering algorithms for e-paper/e-ink displays";
+    homepage = "https://github.com/OpenDisplay/epaper-dithering";
+    changelog = "https://github.com/OpenDisplay/epaper-dithering/releases/tag/${finalAttrs.src.tag}";
+    license = lib.licenses.mit;
+    maintainers = [ lib.maintainers.jamiemagee ];
+  };
+})

--- a/pkgs/development/python-modules/py-opendisplay/default.nix
+++ b/pkgs/development/python-modules/py-opendisplay/default.nix
@@ -1,0 +1,51 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  hatchling,
+  bleak,
+  bleak-retry-connector,
+  epaper-dithering,
+  numpy,
+  pillow,
+  pytestCheckHook,
+  pytest-asyncio,
+}:
+
+buildPythonPackage (finalAttrs: {
+  pname = "py-opendisplay";
+  version = "5.5.0";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "OpenDisplay";
+    repo = "py-opendisplay";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-pPV4Ir9GK++66qq5QGnwyjpBinK7EvI7C7HB14tFDXU=";
+  };
+
+  build-system = [ hatchling ];
+
+  dependencies = [
+    bleak
+    bleak-retry-connector
+    epaper-dithering
+    numpy
+    pillow
+  ];
+
+  nativeCheckInputs = [
+    pytestCheckHook
+    pytest-asyncio
+  ];
+
+  pythonImportsCheck = [ "opendisplay" ];
+
+  meta = {
+    description = "Python library for communicating with OpenDisplay BLE e-paper displays";
+    homepage = "https://github.com/OpenDisplay/py-opendisplay";
+    changelog = "https://github.com/OpenDisplay/py-opendisplay/releases/tag/${finalAttrs.src.tag}";
+    license = lib.licenses.mit;
+    maintainers = [ lib.maintainers.jamiemagee ];
+  };
+})

--- a/pkgs/servers/home-assistant/component-packages.nix
+++ b/pkgs/servers/home-assistant/component-packages.nix
@@ -4535,11 +4535,12 @@
         home-assistant-intents
         ifaddr
         mutagen
+        py-opendisplay
         pymicro-vad
         pyserial
         pyspeex-noise
         zeroconf
-      ]; # missing inputs: py-opendisplay
+      ];
     "openerz" =
       ps: with ps; [
         openerz-api
@@ -8016,6 +8017,7 @@
     "open_router"
     "openai_conversation"
     "openalpr_cloud"
+    "opendisplay"
     "openerz"
     "openevse"
     "openexchangerates"

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -5054,6 +5054,8 @@ self: super: with self; {
 
   enzyme = callPackage ../development/python-modules/enzyme { };
 
+  epaper-dithering = callPackage ../development/python-modules/epaper-dithering { };
+
   epc = callPackage ../development/python-modules/epc { };
 
   ephem = callPackage ../development/python-modules/ephem { };
@@ -13144,6 +13146,8 @@ self: super: with self; {
   py-nymta = callPackage ../development/python-modules/py-nymta { };
 
   py-ocsf-models = callPackage ../development/python-modules/py-ocsf-models { };
+
+  py-opendisplay = callPackage ../development/python-modules/py-opendisplay { };
 
   py-opensonic = callPackage ../development/python-modules/py-opensonic { };
 

